pkg/apis/core/validation/validation.go

```diff
@@ -5941,6 +5942,57 @@ func validateWindowsSecurityContextOptions(windowsOptions *core.WindowsSecurityC
 	return allErrs
 }
 
+func validateWindowsHostProcessPod(podSpec *core.PodSpec, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	containerCount := 0
+	hostProcessContainerCount := 0
+
+	podHostProcess := false
+	if podSpec.SecurityContext != nil && podSpec.SecurityContext.WindowsOptions != nil && podSpec.SecurityContext.WindowsOptions.HostProcess != nil {
+		podHostProcess = *podSpec.SecurityContext.WindowsOptions.HostProcess
```

Comment:
I think it should - yes.
I am trying to follow to same inheritance logic as the other fields in WindowsSecurityContextOptions.
If the pod sets `RunAsUseraname : user1` but then all containers set `RunAsUsername: user2` then all containers would run as `user2`.

Reply:
Let me add a comment to add some clarity!

Reply:
Actually, I just checked against the KEP and we do want to fail validation if `pod hostProcess: false` but any containers set `hostProcess: true`

https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support#kubernetes-api-updates

> If PodSecurityContext.WindowsSecurityContextOptions.HostProcess = false no containers may set Container.SecurityContext.WindowsSecurityContextOptions.HostProcess = true

Reply:
is there a difference of explicit `false` and not set?

Reply:
Yes. If the value is not set at the pod-level we'll allow containers to be scheduled as `HostProcess` containers if all containers set the field.
If the value is set to false explicitly at the pod-level we'll fail validation.

Part of this behavior is forward-looking since we expect to be able to support pods with mixed `HostProcess` and `non-HostProcess` containers with some OS updates.
The other part of this behavior was to give cluster operators an easy way to author policies that forbid HostProcess containers. We see some practical use cases where HostProcess containers will be allowed in some pod namespaces like kube-system but forbidden from others.

Reply:
I'll add comments to clarify this

Reply:
Updated the validation logic in both kube-apiserver and kubelet and also added unit test cases in each to ensure if the Pod's WindowsOptions.HostProcess is explicitly set to false then the pod will not validate/schedule.